code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -210,9 +213,12 @@
 		var/datum/reagent/temp = GLOB.chemical_reagents_list[re]
 		if(temp)
 			var/chemname = temp.name
-			if(is_hallucinating && prob(5))
-				chemname = "[pick_list_replacements("hallucination.json", "chemicals")]"
-			chemicals.Add(list(list("title" = chemname, "id" = ckey(temp.name))))
+			if(HAS_TRAIT(user, TRAIT_CHEMISTRY))
+				if(is_hallucinating && prob(5))
+					chemname = "[pick_list_replacements("hallucination.json", "chemicals")]"
+				chemicals.Add(list(list("title" = chemname, "id" = ckey(temp.name))))
+			else
+				chemicals.Add(list(list("title" = "", "id" = ckey(temp.name))))
```

Comment:
Same as the other one.

Reply:
Algorithm for this changed. No longer relevant.